packages/atomic/src/components/result-template-components/atomic-size-condition/atomic-size-condition.tsx

```diff
@@ -0,0 +1,150 @@
+import {Element, Component, Prop, State} from '@stencil/core';
+
+type Criterion = (width: number, height: number) => boolean;
+
+/**
+ * Displays children only if their parent meets the defined size constraints.
+ */
+@Component({
+  tag: 'atomic-size-condition',
+  styleUrl: 'atomic-size-condition.pcss',
+  shadow: false,
+})
+export class AtomicSizeCondition {
+  private resizeObserver!: ResizeObserver;
+
+  @Element() private host!: HTMLElement;
+
+  @State() private shouldRender!: boolean;
+
+  /**
+   * The minimum width (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumWidthExclusive?: string;
+  /**
+   * The minimum width (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumWidthInclusive?: string;
+  /**
+   * The maximum width (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumWidthExclusive?: string;
+  /**
+   * The maximum width (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumWidthInclusive?: string;
+  /**
+   * The minimum height (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumHeightExclusive?: string;
+  /**
+   * The minimum height (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumHeightInclusive?: string;
+  /**
+   * The maximum height (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumHeightExclusive?: string;
+  /**
+   * The maximum height (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumHeightInclusive?: string;
+
+  constructor() {
+    this.shouldRender = this.elementMeetsCriteria;
+  }
+
+  public componentDidLoad() {
```

Comment:
Why not use the connectedCallback & disconnectedCallback ?
Will the content load nevertheless? Let's say I have a huge image that is only displayable on big screens, will it still render on a mobile?

Reply:
Originally I used connectedCallback and disconnectedCallback and changed them after finding a tutorial on using ResizeObserver in Stencil. Turns out I didn't need to change them, so I'll switch to connectedCallback and disconnectedCallback.